OpenFilesScreen.c

```diff
@@ -164,6 +168,9 @@ static OpenFiles_ProcessData* OpenFilesScreen_getProcessData(pid_t pid) {
          {
             size_t index = getIndexForType(cmd);
             free_and_xStrdup(&item->data[index], line + 1);
+            if (strlen(item->data[index]) > (size_t)pdata->cols[index]) {
+               pdata->cols[index] = (int)CLAMP(strlen(item->data[index]), 0, INT16_MAX);
+            }
```

Comment:
```c
            int len = (int)CLAMP(strlen(item->data[index]), 0, INT16_MAX);
            assert(index < ARRAYSIZE(pdata->cols));
            if (len > pdata->cols[index]) {
               pdata->cols[index] = len;
            }
```

Reply:
Not necessary, as the `cols` array just mirrors the number of items from the `data` array, thus `index` always holds the correct value based on `size_t index = getIndexForType(cmd);`

Reply:
@BenBE I just edited the post and corrected that. About the "optimization" I refer to using `strnlen` or `memchr` instead of `strlen` and then `CLAMP`.

Reply:
I applied the `strlen` part of that optimization; neither of the other two suggestions are fully standards compliant, because they imply the buffer is larger than it actually is.